### ISSUE TITLE
Do not create jsconfig.json on startup

### ISSUE THREAD

**Issue by stashymane** [forcedotcom/salesforcedx-vscode#4822]
I currently have a project with a custom `jsconfig.json` in the root project directory (force-app/main/default). It has specific options to allow me to have proper intellisense & automatic imports between different LWC components for *all sub-projects*.
The project layout is this, more or less:

- `force-app/main/default`
  - `lwc/`  - generic components for org
  - `subproject/`  - specific project subfolder for organization
    - `moduleA/lwc/`
    - `moduleB/lwc/`
  - `jsconfig.json`
 
The problem - the SFDX VSCode extension automatically creates default `jsconfig.json` files for **every** sub-project every time VSCode starts, which completely undoes the config I set up in the main project directory.

It does not make sense to create these files on every startup. A decent solution would be to create them only when creating a new subdirectory, or a new project in general. Otherwise I have to duplicate the `jsconfig.json` file in every single subdirectory, which will eventually become unmaintainable. Or, at least a configuration option to disable this feature would work great.

There already are issues regarding this, however they are closed as either resolved or as a duplicate, without an actual resolution:
- #1394 is only regarding ESLint & user settings, not jsconfig
- #1578 is closed as a duplicate, the problem itself is not resolved

TLDR; please allow us to have an option whether to generate the `jsconfig.json` file on startup or not. We cannot have proper IntelliSense any other way, at least currently.

**Comment by stashymane** [comment]
Another issue related to this, if I do set up `jsconfig.json` files for every `lwc` subdirectory, the extension modifies them on startup as well, introducing a few incompatible changes:

1. Includes a `paths` entry in the root object - the `paths` entry should be included in the `compilerOptions` block, not in the root.

![image](https://user-images.githubusercontent.com/4071883/234275259-f5e82d05-e91b-451b-829a-0066155e9e5d.png)

2. Adds an incompatible include path - the first two elements were added by me and are based on the relative path to the current module, to index all components in the entire project. The SFDX typing path is correct and is not regenerated or mangled, however the additional `**/*` makes IntelliSense recurse through the current path twice, as it is already included by the first line. Every time I remove this wildcard it is regenerated on the next start.

![image](https://user-images.githubusercontent.com/4071883/234275361-8752f808-d819-43d6-9d36-c30cfd0610d2.png)

Even if the first point is configured correctly (the `paths` block is moved into `compilerOptions`), one bug remains due to the structure of LWC components - automatic imports will duplicate the component name, as VSCode has no idea how to deal with the folder structure. The resulting import will be `c/componentName/componentName`, as VSCode looks at the component purely as a folder.  
I have resolved this bug by including `index.js` files in every component that re-export every export in the main component, which does not change any functionality in Salesforce, however it lets VSCode realise the entire directory can be counted as a valid import.

The second point is a slightly easier fix, which is either to outright stop modifying the `jsconfig.json` file, or to check whether this path is included by any other `include` entry.

**Comment by zwhip-stripe** [comment]
@stashymane I just discovered the same issue with `paths` recently. You make some good points in general. JSConfig needs to be generated in order to point vscode towards the types generated by the Apex? language server. 

One thing I wanted to mention is the subproject structure you have is not officially supported, but you can have [multiple package directories ](https://developer.salesforce.com/docs/atlas.en-us.sfdx_dev.meta/sfdx_dev/sfdx_dev_ws_mpd.htm). But -- since you're breaking away from the convention with your structure, is there any real need to having the subfolder of the subprojects be named `lwc`? The extension may leave you alone there if you name it to `components` or something else.

**Comment by zerkz** [comment]
@stashymane I just discovered the same issue with `paths` recently. You make some good points in general. JSConfig needs to be generated in order to point vscode towards the types generated by the Apex? language server. 

One thing I wanted to mention is the subproject structure you have is not officially supported, but you can have [multiple package directories ](https://developer.salesforce.com/docs/atlas.en-us.sfdx_dev.meta/sfdx_dev/sfdx_dev_ws_mpd.htm). But -- since you're breaking away from the convention with your structure, is there any real need to having the subfolder of the subprojects be named `lwc`? The extension may leave you alone there if you name it to `components` or something else.

**Comment by stashymane** [comment]
Yes, the subproject structure is definitely non-standard, but if I made multiple project directories, I would still have to include every other project in each jsconfig as well, though with less nesting. If the jsconfig file wouldn't regenerate, the current subproject structure wouldn't have any issues at all.
I have not tried renaming the folders to something other than LWC yet, will attempt later.

**Comment by rahulgawale** [comment]
I am trying to use tsconfig.json and the jsconfig.json is annoying me. Need a way to "Do not regenerate" and "Do not overwrite" flags for the extension. The auto-create annoys me. Has anyone find any workaround to suppress it?

**Comment by stashymane** [comment]
From what I can remember, you can also regenerate the jsconfig with a sf cli plugin, as I had a somewhat-working POC at some point. That is long gone however, as I no longer work with Salesforce.
